src/main/java/htsjdk/samtools/SamReaderFactory.java

```diff
@@ -74,11 +76,13 @@
 public abstract class SamReaderFactory {
 
     private static ValidationStringency defaultValidationStringency = ValidationStringency.DEFAULT_STRINGENCY;
-    
+
+    protected Function<SeekableByteChannel, SeekableByteChannel> pathWrapper;
+
     abstract public SamReader open(final File file);
 
     public SamReader open(final Path path) {
-        final SamInputResource r = SamInputResource.of(path);
+        final SamInputResource r = SamInputResource.of(path, getPathWrapper());
         final Path indexMaybe = SamFiles.findIndex(path);
         if (indexMaybe != null) r.index(indexMaybe);
```

Comment:
It's up to user code; reading the whole thing is the right thing to do so I expect performance-oriented code will do that already (by "we" earlier I meant my own code).